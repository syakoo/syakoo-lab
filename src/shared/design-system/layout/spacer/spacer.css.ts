import { styleVariants } from "@vanilla-extract/css";
import { recipe } from "@vanilla-extract/recipes";

import { theme } from "@/shared/design-system/theme.css";

export const spacerStyle = recipe({
  variants: {
    x: styleVariants(theme.space, (value) => ({
      width: value,
    })),
    y: styleVariants(theme.space, (value) => ({
      height: value,
    })),
  },
});
