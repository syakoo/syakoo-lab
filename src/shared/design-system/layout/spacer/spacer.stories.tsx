import type { Meta, StoryObj } from "@storybook/nextjs";

import { theme } from "@/shared/design-system/theme.css";

import { Spacer } from ".";

const meta = {
  component: Spacer,
  parameters: {
    layout: "fullscreen",
    testLevel: "snapshot",
  },
  tags: ["autodocs"],
} satisfies Meta<typeof Spacer>;

export default meta;

const DummyBlock = ({ width, height }: { width: string; height: string }) => (
  <div
    style={{
      borderRadius: theme.radius[100],
      background: theme.color.palette.gray[100],
      width,
      height,
      display: "flex",
      alignItems: "center",
      justifyContent: "center",
    }}
  />
);

type Story = StoryObj<typeof meta>;
export const X: Story = {
  args: { x: "200" },
  render: (props) => {
    return (
      <div style={{ display: "flex" }}>
        <DummyBlock height="50px" width="50px" />
        <Spacer {...props} />
        <DummyBlock height="50px" width="50px" />
      </div>
    );
  },
};

export const Y: Story = {
  args: { y: "200" },
  render: (props) => {
    return (
      <div>
        <DummyBlock height="50px" width="50px" />
        <Spacer {...props} />
        <DummyBlock height="50px" width="50px" />
      </div>
    );
  },
};
