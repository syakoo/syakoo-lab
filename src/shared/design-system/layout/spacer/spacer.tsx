import type { RecipeVariants } from "@vanilla-extract/recipes";

import { spacerStyle } from "./spacer.css";

type SpacerProps = RecipeVariants<typeof spacerStyle>;

export const Spacer: React.FC<SpacerProps> = (props) => {
  return <div aria-hidden className={spacerStyle(props)} />;
};
