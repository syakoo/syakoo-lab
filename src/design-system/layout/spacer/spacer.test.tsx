import { testStories } from "@/test/test-stories";

import * as stories from "./spacer.stories";

describe("Spacer", () => {
  testStories(stories);
});
