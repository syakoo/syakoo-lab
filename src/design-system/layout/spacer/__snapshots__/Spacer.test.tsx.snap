// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spacer スナップショット X 1`] = `
<body>
  <div>
    <div
      style="display: flex;"
    >
      <div
        style="border-radius: var(--radius-100__13jk9q11g); width: 50px; height: 50px; display: flex; align-items: center; justify-content: center;"
      />
      <div
        aria-hidden="true"
        class="spacer__1k4u8xae spacer_spacerStyle_x_200__1k4u8xai spacer_spacerStyle_variants_x_200__1k4u8xa3"
      />
      <div
        style="border-radius: var(--radius-100__13jk9q11g); width: 50px; height: 50px; display: flex; align-items: center; justify-content: center;"
      />
    </div>
  </div>
</body>
`;

exports[`Spacer スナップショット Y 1`] = `
<body>
  <div>
    <div>
      <div
        style="border-radius: var(--radius-100__13jk9q11g); width: 50px; height: 50px; display: flex; align-items: center; justify-content: center;"
      />
      <div
        aria-hidden="true"
        class="spacer__1k4u8xae spacer_spacerStyle_y_200__1k4u8xap spacer_spacerStyle_variants_y_200__1k4u8xaa"
      />
      <div
        style="border-radius: var(--radius-100__13jk9q11g); width: 50px; height: 50px; display: flex; align-items: center; justify-content: center;"
      />
    </div>
  </div>
</body>
`;
